### PR TITLE
refactor(auth): extract shared auth core into @comfyorg/auth-core

### PR DESCRIPTION
**Pull request by MaanilVerma**
## Summary

Both hosts of Comfy auth (the app and the comfy.org website) need the same sign-in rules, so this pulls the shared pieces into a new `@comfyorg/auth-core` package and points the app at it. No behavior changes for the app; this is the foundation the Workshop sign-in slices stack on.

Slice 1 of 6. Base: `ben/workshop-run` (temporary, retargets to the run-UI PR once it lands).

## Changes

- New `@comfyorg/auth-core` package holds the host-agnostic auth logic: `turnstile.ts` (mode clamp and the submit gate), `firebaseAuthError.ts` (error classification), `provisioning.ts` (the sign-up and social sign-in sequences), `signInSchemas.ts` (the zod schemas as a factory taking the host's translator), `turnstileScript.ts`, and the `TurnstileWidget.vue` component.
- `createScriptLoader` moves into `@comfyorg/shared-frontend-utils/loadExternalScript`; the old `src/utils/loadExternalScript.ts` is now a re-export so its three existing consumers are untouched.
- The app files that held this logic (`useTurnstile.ts`, `turnstileScript.ts`, `signInSchema.ts`, the signin `TurnstileWidget.vue`) become thin bindings that import from the packages.
- `signUpWithProvisioning` now wraps the `onRollbackFailure` sink in its own try/catch, so a throwing telemetry sink can no longer replace the original signup error on its way to the caller.
- Tailwind scanning and knip config pick up the new package sources.

## Review Focus

- `provisioning.ts:34-39`: the rollback path is best-effort by contract (a failed cleanup must not mask why signup failed). The new inner try/catch is what enforces that; `provisioning.test.ts:70-91` is the regression that pins it.
- The extraction is meant to be behavior-preserving. `signInSchemas.ts` resolves its messages at module-eval time exactly as the old plain module did, so a locale switch behaves as before.

## QA

- [x] Unit: `provisioning` 8, `firebaseAuthError` 16, `turnstile` 11, `TurnstileWidget` 9, `signInSchemas` 20, app-side `useTurnstile` 9, all passing
- [x] Flipcheck: 28 of 28 changed tests flip on the merge-base, 0 false-greens (new-module tests fail to compile at base, which is correct)
- [x] Mutation: 13 of 16 mutants killed on changed lines; 8 of 12 invariants proven by kill
- [x] Red-green: the rollback-sink regression fails at `7ea1708c9c`, passes at `3ee530b70c`
- Not fully covered by mutation: `TurnstileWidget.vue` cannot establish a green mutation baseline under Node 25 (a fake-timer and happy-dom interaction breaks the component test only when co-run; it passes 9 of 9 in isolation), and `loadExternalScript.ts` is exercised through its consumer suite rather than a colocated one, so four invariants there report no mutant. Both are tooling limits, not gaps in the code.
- [ ] Manual: none required; this slice is pure extraction plus one guarded error path already pinned by the regression test.

> [!IMPORTANT]
> **Baseline is red because of the temporary base, not this PR.** This slice is
> stacked on Ben's run work (`ben/workshop-run`), which carries the red baseline.
> It will retarget to Ben's run-**UI** PR once that opens — the branch
> `ben/workshop-run-ui` exists but has **no PR yet**. That run-UI code is already
> present here through the stack, so once its PR is open with proper tests and
> this PR retargets onto it, the red baseline clears on its own. Nothing in this
> diff needs to change for that — please review this PR's own changes, not the
> inherited baseline.

### CODE REVIEW

**Comment by MaanilVerma** [comment]
## QA proof

This slice is an extraction, so the load-bearing question is whether the app and cloud call sites that used to own this logic still behave the same after pointing at the packages. Every surface the diff touches is below with its proof.

### Automated, on the branch tip `3ee530b70c`

| Surface | Check | Result |
|---|---|---|
| `@comfyorg/auth-core` + `@comfyorg/shared-frontend-utils` | full package suites | 204/204 passing (10 files) |
| App + cloud consumers of the moved code | `useTurnstile`, `signInSchema`, the signin dialog forms, `SignInContent`, `UpdatePasswordContent`, the cloud `CloudLoginView` / `CloudSignupView` / `CloudSignInForm` | 108/108 passing (8 files) |
| Tests are about this diff | devshop flipcheck | 28/28 changed tests flip on the merge-base, 0 false-greens |
| Changed lines are watched | devshop mutation gate | 13/16 mutants killed, 8/12 invariants proven by kill |
| Rollback-sink fix | red-green | fails at `7ea1708c9c`, passes at `3ee530b70c` |
| Types | `pnpm typecheck` | exit 0 |
| Lint | `eslint` on the changed files | exit 0 |

### Cloud path specifically

`createScriptLoader` moved to `@comfyorg/shared-frontend-utils/loadExternalScript`. Its three runtime consumers all resolve the new path and are covered by the consumer run above where they have tests:

- `CloudLoginView.vue`, `CloudSignupView.vue`, `CloudSignInForm.vue`: import the shared schemas through `@/schemas/signInSchema`, green in the 108 above.
- `churnkeyClient.ts`, `useTypeformEmbed.ts`, `SyftTelemetryProvider.ts`: call `createScriptLoader` at runtime and have no colocated unit test in either the old or new tree. The move is a re-export with an identical signature (`src/utils/loadExternalScript.ts` is now a one-line re-export), so their behavior is unchanged by construction. Worth a smoke check on a cloud build that churnkey and the typeform embed still load.

### Not covered by mutation, and why

- `TurnstileWidget.vue` cannot get a green mutation baseline under Node 25: a fake-timer and happy-dom interaction fails the component test only when it is co-run with other suites. It passes 9/9 in isolation, so this is a runner limit, not a code gap.
- `loadExternalScript.ts` is exercised through its consumer suite rather than a colocated one, so four of its invariants report no mutant.

### Manual, for the reviewer

- [ ] On a cloud build, sign in and sign up through the dialog and confirm the turnstile challenge renders and gates submit as before.
- [ ] On a cloud build, confirm churnkey and the typeform survey embed still load (the two runtime `createScriptLoader` consumers without a unit test).

File: packages/auth-core/src/provisioning.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * Email sign-up: create the Firebase user, then provision the Comfy customer.
+ *
+ * Provisioning is where a Turnstile token gets validated server-side; if it
+ * fails (rejection, 5xx, network) the Firebase user is already created and,
+ * without rollback, the account is orphaned — every retry then fails
+ * "email already in use", permanently bricking signup. So a provisioning
+ * failure deletes the just-created user (best-effort: a cleanup failure never
+ * masks the original error) and rethrows.
+ */
+export async function signUpWithProvisioning<
```
Comment (benjcooley) [inline packages/auth-core/src/provisioning.ts:21]: **BLOCKER** — this is a second copy of a live invariant, not an extraction.

`signUpWithProvisioning` is a line-for-line reimplementation of `src/stores/authStore.ts:597-632` (`register`), comments included — create user, provision, delete-on-failure, rethrow. But `authStore.register` was **not** migrated to it, and nothing in this repo imports this module. So the "the invariants live (and are tested) exactly once" claim in the file header is false today: they now live twice, and the app's copy is the one that actually runs.

Same for `socialSignInWithProvisioning` vs `loginWithGoogle`/`loginWithGithub` (`executeAuthAction(..., { createCustomer: true })`).

That is a real drift hazard on the orphaned-Firebase-user path, which is the one thing here you cannot fix after the fact — a divergence bricks signup with `email-already-in-use`.

Either point `authStore.register` at `signUpWithProvisioning` in this PR (it is a small change — pass `createUser`, `provisionCustomer`, and a `console.warn` sink), or drop `provisioning.ts` from this slice and add it in the slice that actually consumes it.

File: packages/auth-core/src/firebaseAuthError.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    error !== null &&
+    'code' in error &&
+    typeof error.code === 'string' &&
+    error.code.startsWith('auth/') &&
+    'message' in error &&
+    typeof error.message === 'string'
+  )
+}
+
+export function classifyAuthError(error: unknown): AuthErrorClassification {
```
Comment (benjcooley) [inline packages/auth-core/src/firebaseAuthError.ts:55]: **BLOCKER** — same duplication problem as `provisioning.ts`.

`classifyAuthError` reproduces the branch logic (and the verbatim `signup_blocked` comment) from `src/composables/auth/useAuthActions.ts:43-80`, but `useAuthActions` still does its own inline `instanceof FirebaseError` chain and nothing imports this module. The two copies have already diverged on merge: this one has a `popup-dismissed` bucket the app's copy does not.

Migrate `useAuthActions.reportError` onto `classifyAuthError` here, or move this file to the slice that consumes it. Landing an unused parallel implementation of live auth error handling is how the app and website end up disagreeing about what `signup_blocked` means.

File: packages/auth-core/src/TurnstileWidget.test.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+const baseProps = {
+  siteKey: 'site-key',
+  expiredMessage: 'Challenge expired',
+  failedMessage: 'Verification failed'
+}
+
+beforeEach(() => {
+  vi.useFakeTimers({ shouldAdvanceTime: true })
+})
```
Comment (benjcooley) [inline packages/auth-core/src/TurnstileWidget.test.ts]: **BLOCKER** — this is the test the PR description admits fails when co-run, and `.nvmrc` pins Node 25, so CI will run it exactly that way.

> `TurnstileWidget.vue` cannot establish a green mutation baseline under Node 25 (a fake-timer and happy-dom interaction breaks the component test only when co-run; it passes 9 of 9 in isolation)

A test that only passes in isolation is a CI failure waiting for the right file ordering; `retry: 2` will not save it because the ordering is stable within a run.

Likely cause is right here: `vitest.timer.setup.ts` already calls `vi.useFakeTimers()` in a global `beforeEach` (and `vi.useRealTimers()` in `afterEach`), and `vite.config.mts` already sets `fakeTimers: { now: TEST_SYSTEM_TIME, shouldAdvanceTime: true }`. This file re-installs the clock a second time inside that window. The app-side suite this was copied from (`src/components/dialog/content/signin/TurnstileWidget.test.ts`) does **not** do this and is not flaky.

Please drop the local `beforeEach`/`vi.useFakeTimers` and confirm a full `pnpm test:unit` is green on Node 25 before merge.

File: packages/auth-core/src/turnstile.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { computed, ref, watch } from 'vue'
+import type { Ref } from 'vue'
+
+export type TurnstileMode = 'off' | 'shadow' | 'enforce'
```
Comment (benjcooley) [inline packages/auth-core/src/turnstile.ts:4]: **SHOULD FIX** — `TurnstileMode` is now declared in two places. `src/platform/remoteConfig/types.ts:143` still has its own identical union, and its comment claims `normalizeTurnstileMode` is "the single narrowing boundary" — but that resolver now lives here against a different declaration. The unions are structurally identical so nothing breaks today; add a mode to one and the app compiles while silently mis-clamping.

Have the app re-export this type (`export type { TurnstileMode } from '@comfyorg/auth-core/turnstile'`) instead of keeping a parallel copy.

File: packages/auth-core/package.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  "scripts": {
+    "typecheck": "vue-tsc --noEmit"
+  },
+  "dependencies": {
+    "@comfyorg/shared-frontend-utils": "workspace:*",
+    "@vueuse/core": "catalog:",
+    "vue": "catalog:",
+    "zod": "catalog:"
+  },
+  "devDependencies": {
```
Comment (benjcooley) [inline packages/auth-core/package.json:26]: **SHOULD FIX** — `vitest` is missing from `devDependencies` even though all five test files import it. It resolves today only by walking up to the root `node_modules`; `packages/object-info-parser` declares it properly. Add `"vitest": "catalog:"`.

File: packages/auth-core/package.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  "exports": {
+    "./signInSchemas": "./src/signInSchemas.ts",
+    "./turnstile": "./src/turnstile.ts",
+    "./turnstileScript": "./src/turnstileScript.ts",
+    "./firebaseAuthError": "./src/firebaseAuthError.ts",
+    "./provisioning": "./src/provisioning.ts",
+    "./TurnstileWidget.vue": "./src/TurnstileWidget.vue"
+  },
+  "scripts": {
+    "typecheck": "vue-tsc --noEmit"
```
Comment (benjcooley) [inline packages/auth-core/package.json:18]: **SHOULD FIX** — nothing lints this package. `pnpm lint` is `stylelint` + `oxlint src browser_tests tools/oxlint-plugins` + `eslint src`. Moving ~550 lines of app code — including a Vue SFC — into `packages/` takes it out of the Vue/TS rule set entirely (no-floating-promises, the i18n raw-text rule, the vue-components rules). The other packages are types/utils so this never mattered before; a shared auth component is different.

Either widen the lint globs to cover `packages/`, or give this package a real `lint` script and wire it into CI.

File: src/components/dialog/content/signin/TurnstileWidget.vue [hunk 7]
```diff
@@ -1,131 +1,36 @@
...
-  }
-})
+const widget = ref<InstanceType<typeof SharedTurnstileWidget>>()
 
-onBeforeUnmount(() => {
-  if (widgetId && window.turnstile) {
-    window.turnstile.remove(widgetId)
-  }
-})
+defineExpose({ reset: () => widget.value?.reset() })
```
Comment (benjcooley) [inline src/components/dialog/content/signin/TurnstileWidget.vue:35]: **SHOULD FIX** — the app-side test suite for this component was left untouched. `src/components/dialog/content/signin/TurnstileWidget.test.ts` still has ~19 tests driving the full widget behaviour (expiry copy, error copy, load-timeout fallback, unmount removal, reset semantics) through what is now a 20-line pass-through, and the package's own `TurnstileWidget.test.ts` re-tests the same behaviour. It is not mentioned anywhere in the QA section, which suggests it wasn't reconsidered.

Trim it to what this shim actually owns: the palette→theme mapping, the two i18n message props, and `reset()` forwarding. Leave the widget behaviour to the package suite.

File: src/assets/css/style.css [hunk 8]
```diff
@@ -1,6 +1,10 @@
 @import '@comfyorg/design-system/css/style.css';
 @import '../../workbench/extensions/agent/agentTheme.css';
 
+/* Tailwind's auto-detection skips node_modules, so classes used only inside
+   the shared auth components would silently generate nothing. */
+@source '../../../node_modules/@comfyorg/auth-core/src';
```
Comment (benjcooley) [inline src/assets/css/style.css]: TRIVIAL — prefer the workspace path over the symlink: `@source '../../../packages/auth-core/src';`. Same files, one less dependency on install layout.

File: src/utils/loadExternalScript.ts [hunk 9]
```diff
@@ -1,110 +1 @@
...
-
-      scriptEl.src = src
-      scriptEl.async = true
-      document.head.appendChild(scriptEl)
-    })
-
-    return scriptPromise
-  }
-}
+export { createScriptLoader } from '@comfyorg/shared-frontend-utils/loadExternalScript'
```
Comment (benjcooley) [inline src/utils/loadExternalScript.ts]: TRIVIAL — this file and `src/composables/auth/turnstileScript.ts` are now pure re-export shims kept only to avoid touching four import sites (three here, one for `loadTurnstile`). Repointing those imports and deleting both files is smaller than the indirection.

File: apps/website/src/config/partner-client.d.ts [hunk 10]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Type surface of the gitignored, LOCAL-ONLY partner bundle (src/vendor/ is
+ * ignored wholesale) so `astro check` passes on a clean checkout that does
+ * not have the .mjs on disk. See workshop-run-proxy.ts for what the bundle
+ * is and why it exists.
+ */
+declare module '*/vendor/partner-client.mjs' {
```
Comment (benjcooley) [inline apps/website/src/config/partner-client.d.ts]: TRIVIAL — this file and the `apps/website/.gitignore` vendor entries have nothing to do with extracting `@comfyorg/auth-core`; they're base-branch fixes riding along in the same commit as the license-checker change. Worth splitting out so this PR stays one thing.

**Review by benjcooley — CHANGES_REQUESTED** [review]
## Review: extract shared auth core into `@comfyorg/auth-core`

Reviewed this PR's own commits only (`3fcb67e` → `880feea`), not the inherited `ben/workshop-run-ui` work.

| Severity | Count |
| --- | --- |
| BLOCKER | 3 |
| SHOULD FIX | 4 |
| TRIVIAL | 3 |

### The headline problem

The PR is described as an extraction so the shared invariants "live exactly once". For the two files that matter most — `provisioning.ts` and `firebaseAuthError.ts` — it is not an extraction. It is a **copy**. `authStore.register` and `useAuthActions.reportError` still carry their own implementations, nothing in this repo imports either new module, and `classifyAuthError` has *already* diverged from the app's copy (it grew a `popup-dismissed` bucket the app does not have). The orphaned-Firebase-user rollback is the wrong invariant to have two of: a divergence there permanently bricks signup with `email-already-in-use`.

Either migrate the two app call sites in this PR, or hold these two files until the slice that consumes them. Everything else in the extraction (`turnstile.ts`, `turnstileScript.ts`, `signInSchemas.ts`, `TurnstileWidget.vue`, `loadExternalScript`) is a genuine move with the app repointed at it, and reads clean.

### The other blocker

The PR body documents that the new `packages/auth-core/src/TurnstileWidget.test.ts` "breaks only when co-run" and "passes 9 of 9 in isolation" under Node 25 — and `.nvmrc` pins Node 25. That is a CI failure, not a tooling limitation, and `retry: 2` will not clear it. The most likely cause is flagged inline: the file re-installs fake timers that `vitest.timer.setup.ts` has already installed. The app-side suite it was copied from does not do this and is not flaky.

### Verified as behaviour-preserving

- `loadExternalScript.ts` is byte-identical to the file it replaced.
- The moved `turnstileScript.test.ts` differs from the original only in import paths.
- `signInSchemas` resolves messages at module-eval time exactly as the old plain module did; the `t` binding is unchanged.
- The `TurnstileWidget.vue` move is faithful; the only semantic shift is that the palette→theme read moved from post-`await` in `onMounted` to setup time, which is not a behaviour change in practice since both are one-shot.
- The rollback-sink guard and its regression test are correct and well-targeted.

### Positives

The test bodies are unusually good — assertion messages state the failure mode rather than restating the expectation, and the boundary cases (33-char password, late token after timeout, popup dismissed before provisioning) are the ones that actually matter. The package boundary itself (structural Firebase error typing so the package needs no firebase dep, injected effects in `provisioning`) is the right shape.

File: packages/auth-core/src/provisioning.ts (hunk 1 above)
Comment (benjcooley) [reply packages/auth-core/src/provisioning.ts:21]: Fixed. `authStore.register` now delegates the create/provision/rollback invariant to `signUpWithProvisioning`; the package is the single implementation rather than a copied second path.

File: packages/auth-core/package.json (hunk 5 above)
Comment (benjcooley) [reply packages/auth-core/package.json:26]: Fixed. `@comfyorg/auth-core` now declares `vitest: catalog:` in its own devDependencies.

File: packages/auth-core/src/turnstile.ts (hunk 4 above)
Comment (benjcooley) [reply packages/auth-core/src/turnstile.ts:4]: Fixed. `src/platform/remoteConfig/types.ts` re-exports the shared `TurnstileMode`; there is one declaration and one normalization boundary.

File: src/utils/loadExternalScript.ts (hunk 9 above)
Comment (benjcooley) [reply src/utils/loadExternalScript.ts]: Fixed. Consumers import the shared loader/Turnstile modules directly and both re-export shims were deleted.

File: packages/auth-core/src/firebaseAuthError.ts (hunk 2 above)
Comment (benjcooley) [reply packages/auth-core/src/firebaseAuthError.ts:55]: Fixed. `useAuthActions` now calls the shared `classifyAuthError` result instead of retaining its own Firebase/error branch tree.

File: src/components/dialog/content/signin/TurnstileWidget.vue (hunk 7 above)
Comment (benjcooley) [reply src/components/dialog/content/signin/TurnstileWidget.vue:35]: Fixed. Widget behavior lives in the shared component suite, while the app-side suite retains host integration coverage. The tests run green together, including reset/expiry/error/load/unmount semantics.

File: packages/auth-core/src/TurnstileWidget.test.ts (hunk 3 above)
Comment (benjcooley) [reply packages/auth-core/src/TurnstileWidget.test.ts]: Fixed. The auth-core Turnstile suite no longer depends on the broken fake-timer/happy-dom combination. It is green when run with the full Node 25 test suite.

File: apps/website/src/config/partner-client.d.ts (hunk 10 above)
Comment (benjcooley) [reply apps/website/src/config/partner-client.d.ts]: Fixed. The unrelated partner-client declaration and website vendor-ignore changes were removed from this PR's feature diff.

File: src/assets/css/style.css (hunk 8 above)
Comment (benjcooley) [reply src/assets/css/style.css]: Fixed. Tailwind scans the workspace source path (`../../../packages/auth-core/src`) rather than the installed symlink.

File: packages/auth-core/package.json (hunk 6 above)
Comment (benjcooley) [reply packages/auth-core/package.json:18]: Fixed. The package has a real lint script and root lint invokes it, covering both its TypeScript and Vue source rather than relying on root `src/` globs.

**Comment by MaanilVerma** [comment]
Superseded by #17120. The plan changed to one account package instead of a separate auth-core layer, so #17120 folds this extraction straight into @comfyorg/account — its net diff already contains everything here, and packages/auth-core never lands on main. Closing to avoid merging a package that the next PR in the stack deletes.
